hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -375,26 +378,31 @@ private IterationResult doIteration() {
     findSourceStrategy.reInitialize(getPotentialSources(), config, lowerLimit);
     List<DatanodeUsageInfo> potentialTargets = getPotentialTargets();
     findTargetStrategy.reInitialize(potentialTargets, config, upperLimit);
-
     Set<DatanodeDetails> selectedTargets =
         new HashSet<>(potentialTargets.size());
     moveSelectionToFutureMap = new HashMap<>(unBalancedNodes.size());
     boolean isMoveGenerated = false;
+    iterationResult = IterationResult.ITERATION_COMPLETED;
+
     try {
       // match each overUtilized node with a target
       while (true) {
         if (!isBalancerRunning()) {
-          return IterationResult.ITERATION_INTERRUPTED;
+          iterationResult = IterationResult.ITERATION_INTERRUPTED;
+          break;
         }
 
-        IterationResult result = checkConditionsForBalancing();
-        if (result != null) {
-          return result;
+        if (checkConditionsForBalancing() != null) {
```

Comment:
I think the `maxSizeToMovePerIteration` and `maxDatanodesPercentageToInvolvePerIteration` are more like ITERATION_STOP_REASON other than ITERATION_RESULT.
Maybe we can add a new field noting the reason and mark the result as COMPLETE if encounter a valid STOP_REASON?